client/my-sites/comments/comment-list/index.jsx

```diff
@@ -542,8 +542,10 @@ export class CommentList extends Component {
 }
 
 const mapStateToProps = ( state, { postId, siteId, status } ) => {
-	const siteCommentsTree = getSiteCommentsTree( state, siteId, status );
 	const isPostView = !! postId;
+	const siteCommentsTree = isPostView
+		? filter( getSiteCommentsTree( state, siteId, status ), { commentParentId: 0 } )
```

Comment:
One corner case is for pending comments. If we have an approved parent comment, but a nested unapproved comment, it will not display in the 'pending' filter.